### ISSUE TITLE
nagios.cmd is deleted permanently after "service nagios restart" and "service nagios reload"

### ISSUE THREAD

**Issue by PepeOnAChair** [NagiosEnterprises/nagioscore#495]
After upgrade to Nagios 4.3.4, these 2 commands ("service nagios restart" and "service nagios reload") permanently delete "/usr/local/nagios/var/rw/nagios.cmd" pipe.

I need to reload Nagios configuration from cmd by bash scripts. The only way better say work-aroud I found is this set of commans:
service nagios stop ; killall nagios ; sleep 4 ; service nagios start
- with this way of "reloading" Nagios, nagios.cmd pipe is recreated

But this is not nice way of reloading the Nagios because Nagvis map (connected to Nagios with /usr/local/nagios/var/rw/live) loose "connection" with Nagios and avery icons got blue for a few seconds - which is annoying for our customer-support people...

So it seems as a bug to me.
Does anybody know, how to fix this?

Thank you
Pepe

**Comment by ericloyd** [comment]
So "service nagios reload" doesn't reload for you?

**Comment by ericloyd** [comment]
"service nagios reload" should not remove nagios.cmd.  We're not running 4.3.4 at this time, but if that's the case, then I'd consider it a bug as well.  In fact, I can't see how Nagios Core would function properly without a nagios.cmd pipe.  After all, the last time nagios.cmd was mentioned in changelog as back in 4.0.8.

**Comment by PepeOnAChair** [comment]
yes, after I run "service nagios reload", it really deletes nagios.cmd and it doesn`t recreate it :(

**Comment by hedenface** [comment]
What operating system, architecture? How did you upgrade/ install?  Through
packages or compilation? Did you use a walkthrough? If so, which one?

**Comment by PepeOnAChair** [comment]
Hello,
I`m running Nagios on Debian wheezy (HP server ProLiant DL380 G5 )

_uname -a
Linux Nagios 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt20-1+deb8u3 (2016-01-17) x86_64 GNU/Linux_

I did upgrade Nagios from 4.2.4 to 4.3.4 yesterday. I always install Nagios from source codes with compilation.

I don`t know, what you mean by "walkthrough".

Pepe

**Comment by ericloyd** [comment]
A "walkthrough" is a specific thing that if you do not know what it is, then you did not do it.  :-)

**Comment by hedenface** [comment]
Any update on the status of this? Did the manual compilation work?

**Comment by PepeOnAChair** [comment]
Hello,
as I`m kind of overloaded with other IT things, I possibly haven`t found the best solution, but it workd just fine.
These steps I`ve tried. All except the last one was dead-end.

1) service nagios stop ; killall nagios ; sleep 4 ; service nagios start --> pipe was recreated, but host and service states are forgotten. So after this kind of reloading of Nagios frequently, we can`t really trust what we see as the plugin result (as it can be older than expected - I mean that we can see the result/output of the plugin valid longer time before Nagios"reload")

2) I also tried to bypass the problem with changing rights and ownership of the pipe, than "service nagios reload" and then put rights and ownership back (so Nagios would not be able to delete the pipe) -> this did not work

3) simililar to number 2), but with renaming the pipe - also did not work as expected

4) I have noticed, that reloading the Nagios still works perfectly if I click to "Process Info" - "Restart the Nagios process". So in Chrome - Developers tools I grabbed what that link does as the curl command. Than I put this very long command to my reload script. And since then reloading works like a charm - host  and service states are not lost, new hosts and services are really added to be monitored and pipe is recreated.

So this was the way to my workaround.
Pep
